pkg/service/HelmAppServiceWrapper.go

```diff
@@ -166,10 +168,19 @@ func (impl *ApplicationServiceServerImpl) GetDesiredManifest(ctx context.Context
 func (impl *ApplicationServiceServerImpl) UninstallRelease(ctx context.Context, in *client.ReleaseIdentifier) (*client.UninstallReleaseResponse, error) {
 	impl.Logger.Infow("Uninstall release request", "clusterName", in.ClusterConfig.ClusterName, "releaseName", in.ReleaseName,
 		"namespace", in.ReleaseNamespace)
-
 	res, err := impl.HelmAppService.UninstallRelease(in)
 	if err != nil {
+		isReleaseInstalled, releaseErr := impl.HelmAppService.IsReleaseInstalled(context.Background(), in)
+		if releaseErr != nil {
+			impl.Logger.Errorw("error in checking if release is installed or not")
+			return nil, status.Error(codes.Internal, releaseErr.Error())
+		}
+		if !isReleaseInstalled {
+			impl.Logger.Errorw("error, no release found", "ReleaseIdentifier", in)
+			return nil, status.Error(codes.NotFound, fmt.Sprintf(" release not found for '%s'", in.ReleaseName))
+		}
```

Comment:
pls put comments here as to why these changes were necessary

Reply:
done